src/ui/map.js

```diff
@@ -2674,6 +2674,13 @@ class Map extends Camera {
             this.style.update(parameters);
         }
 
+        const fogIsTransitioning = this.style && this.style.fog && this.style.fog.hasTransition();
+
+        if (fogIsTransitioning) {
+            this.style._markersNeedUpdate = true;
+            this._sourcesDirty = true;
```

Comment:
I find it a bit strange to flag the sources dirty here so that the test on line 2687 falls through.

What do you think of rewriting that to have a more explicit fog render loop:

```
// Fog related updates
if (this.style && this.style.fog) {
  const fog = this.style.fog;
  if (fog.hasTransition() || this._sourcesDirty) {
    this.painter._updateFog(this.style);
  }
}
```
Then we'd just remove it at line 2689.

Reply:
Hmm… `_updateFog` computes a new `fogCullDistSqr` value, but it's not adequate to actually apply that culling so that the code above would have no effect.

Reply:
I see, so _sourcesDirty is actually something needed when we have the transition on going, right?

Reply:
I believe so, yes. At least for range and color (via opacity). Anything that would trigger a change in tile loading. My initial approach worked in this additional logic rather than setting _sourcesDirty in the render loop, which admittedly seemed a little odd, but the effect seemed entirely equivalent.

Reply:
Ok makes sense, disregard that comment then!